scripts/background/StateManager.js

```diff
@@ -28,10 +29,15 @@ function StateManager() {
 		getSyncedOptions(() => {
 			if (!weatherManager) {
 				weatherManager = new WeatherManager(options.zipCode, options.countryCode);
-				weatherManager.registerChangeCallback(function () {
+				weatherManager.registerChangeCallback(() => {
 					if (!isKK() && isLive()) {
 						let musicAndWeather = getMusicAndWeather();
-						notifyListeners("hourMusic", [timeKeeper.getHour(), musicAndWeather.weather, musicAndWeather.music, false]);
+
+						// Sends a different event on startup to get the "hourMusic" desktop notification.
+						if (startup) {
+							notifyListeners("hourMusic", [timeKeeper.getHour(), musicAndWeather.weather, musicAndWeather.music, false]);
+							startup = false;
+						} else notifyListeners("weatherChange", [timeKeeper.getHour(), musicAndWeather.weather, musicAndWeather.music, false]);
```

Comment:
I see the intention here is to only send `hourMusic` when the extension starts, and if it is stopped and then started again, then send `weatherChange`. Is it ok to set `startup = true` only when the StateManager is created? I genuinely don't know and can't test it right now, but it might be ok because of JS's scopes.

Reply:
Yes, the reason why that's done is because when the extension first boots, it tries to get the live weather information, which is asynchronous. Before it does this, the live weather information is unknown/undefined, so we use this startup check to purposely wait until we try getting our live weather data. If we succeed (or even fail), then this event gets called, and we start the music using the live weather data right off the bat with the correct notification. I'm unsure what issues you think may arise by setting the variable's value only when the StateManager is created, this all works fine.